tests/test_circuit_grid.py

```diff
@@ -0,0 +1,251 @@
+
+        self.assertEqual(self.grid.selected_wire, 0)
+        self.assertEqual(self.grid.selected_column, 0)
+        self.assertEqual(self.grid.xpos, 0)
+        self.assertEqual(self.grid.ypos, 0.7 * WINDOW_HEIGHT)
+
+    def test_highlight_selected_node(self):
+        """
+        Test highlight selected node
+        """
+
+        self.grid.highlight_selected_node(1, 1)
+
+        self.assertEqual(self.grid.selected_wire, 1)
+        self.assertEqual(self.grid.selected_column, 1)
+
+        self.grid.highlight_selected_node(2, 2)
+
+        self.assertEqual(self.grid.selected_wire, 2)
+        self.assertEqual(self.grid.selected_column, 2)
+
+    def test_reset_cursor(self):
+        """
+        Test reset cursor
+        """
+
+        self.grid.highlight_selected_node(2, 2)
+        self.grid.reset_cursor()
+
+        self.assertEqual(self.grid.selected_column, 0)
+        self.assertEqual(self.grid.selected_wire, 0)
+
+    def test_move_to_adjacent_node(self):
+        """
+        Test move cursor to adjacent node
+        """
+
+        self.grid.move_to_adjacent_node(MOVE_RIGHT)
+
+        self.assertEqual(self.grid.selected_column, 1)
+        self.assertEqual(self.grid.selected_wire, 0)
+
+        self.grid.move_to_adjacent_node(MOVE_DOWN)
+
+        self.assertEqual(self.grid.selected_column, 1)
+        self.assertEqual(self.grid.selected_wire, 1)
+
+        self.grid.move_to_adjacent_node(MOVE_LEFT)
+
+        self.assertEqual(self.grid.selected_column, 0)
+        self.assertEqual(self.grid.selected_wire, 1)
+
+        self.grid.move_to_adjacent_node(MOVE_UP)
+
+        self.assertEqual(self.grid.selected_column, 0)
+        self.assertEqual(self.grid.selected_wire, 0)
+
+    def test_get_selected_node_gate_part(self):
+        """
+        Test getting gate on currently selected node
+        """
+
+        node1 = CircuitGridNode(node_types.Y)
+        node2 = CircuitGridNode(node_types.X)
+        node3 = CircuitGridNode(node_types.Z)
+
+        self.circuit_grid_model.set_node(0, 1, node1)
+        self.circuit_grid_model.set_node(1, 0, node2)
+        self.circuit_grid_model.set_node(2, 0, node3)
+
+        self.grid.move_to_adjacent_node(MOVE_RIGHT)
+
+        selected_gate = self.grid.get_selected_node_gate_part()
+        self.assertEqual(node1.node_type, selected_gate)
+
+        self.grid.move_to_adjacent_node(MOVE_LEFT)
+        self.grid.move_to_adjacent_node(MOVE_DOWN)
+
+        selected_gate = self.grid.get_selected_node_gate_part()
+        self.assertEqual(node2.node_type, selected_gate)
+
+        self.grid.move_to_adjacent_node(MOVE_DOWN)
+
+        selected_gate = self.grid.get_selected_node_gate_part()
+        self.assertEqual(node3.node_type, selected_gate)
+
+    def test_handle_single_qubit_gate_inputs(self):
+        """
+        Test handling input placing single
+        qubit gates on circuit grid
+        """
+
+        # 0 X-------
+        # 1 Y------
+        # 2 Z------
+        # 3 H------
```

Comment:
Unify the number of -- after gates

Reply:
The number of ascii dashes on each ascii diagram is now 9.